### PR TITLE
gsl/develop RRFS-Smoke

### PR DESCRIPTION
**Pull request by haiqinli**
1). to include the smoke subroutines;
2). to update MYNN PBL transport, Thompson MP aerosol indirect feedback, radiation direct feedback;
3). Passed the regression test run case of rrfs_conus13km_hrrr_warm with Intel and GNU compiler on Hera;
      The successful log files are at:
      /scratch2/BMC/acomp/Haiqin.Li/RegressionTest/RRFS-Smoke/ufs-weather-model/tests/intel_out
      /scratch2/BMC/acomp/Haiqin.Li/RegressionTest/RRFS-Smoke/ufs-weather-model/tests/gnu_out

4). Created a new regression test run case of /scratch2/BMC/acomp/Haiqin.Li/RegressionTest/RRFS-Smoke/ufs-weather-model/tests/rundir/rrfs_smoke_conus13km_hrrr_warm. The meteorological fields are identical with rrfs_conus13km_hrrr_warm the when turning off direct/indirect aerosol feedback.

### CODE REVIEW

**Comment by SamuelTrahanNOAA** [comment]
You need a pull request at the ufs-weather-model level.

https://github.com/NOAA-GSL/ufs-weather-model/pulls

**Review by mdtoyNOAA — APPROVED** [review]

**Review by hannahcbarnes — APPROVED** [review]

File: ccpp/data/GFS_typedefs.F90 [hunk 1]
```diff
@@ -5471,13 +5608,20 @@ subroutine control_chemistry_initialize(Model, tracer_types)
...
     Model%ntche = NO_TRACER
     Model%ndchm = 0
     Model%ndchs = NO_TRACER
     Model%ndche = NO_TRACER
 
+    if (Model%rrfs_smoke) then
+      Model%nchem = 2
+      Model%ndvel = 2
+    endif
+
```
Comment (joeolson42) [inline ccpp/data/GFS_typedefs.F90:5624]: I must have missed it, but where is mix_chem set to true when rrfs_smoke is set to true? Or do you have to set both manually?

Comment (haiqinli) [reply ccpp/data/GFS_typedefs.F90:5624]: The mix_chem and fire_turb are false by default. Actually all the smoke related options are false by default.

    logical :: smoke_forecast = .false.   ! RRFS-smoke diurnal 
    logical :: aero_ind_fdb = .false.     ! RRFS-smoke wfa/ifa emission
    logical :: aero_dir_fdb = .false.     ! RRFS-smoke smoke/dust radiation feedback
    logical :: rrfs_smoke_debug = .false. ! RRFS-smoke plumerise debug
    logical :: mix_chem = .false.         ! tracer mixing option by MYNN PBL
    logical :: fire_turb = .false.        ! enh vertmix option by MYNN PBL

Then they are read from the external namelist file of gfs_physics_nml. In standard alone RRFS (SDF of FV3_HRRR), they are false. In the RRFS-Smoke (SDF of FV3_HRRR_smoke), they are true.

                          !--- RRFS smoke namelist
                               seas_opt, dust_opt, biomass_burn_opt, drydep_opt,            &
                               wetdep_ls_opt, smoke_forecast, aero_ind_fdb, aero_dir_fdb,   &
                               rrfs_smoke_debug, do_plumerise, plumerisefire_frq,           &
                               addsmoke_flag, fire_turb, mix_chem,                          &

Comment (joeolson42) [reply ccpp/data/GFS_typedefs.F90:5624]: So, I think that means they both need to be set = true in the namelist. 

Anyway, in the interest of time, I will approve these changes, but eventually "kdvel" will have to be specified outside of the MYNN wrapper and hard-coding "EMIS_ANT_NO" and "vdep" should also be removed from the MYNN wrapper. This won't hurt to leave them in the wrapper for now, but this really isn't the proper location to handle these non-mynn-specific variables.

**Review by joeolson42 — APPROVED** [review]
reluctant approval with desire of future clean-up.

File: ccpp/data/GFS_typedefs.F90 (hunk 1 above)
Comment (haiqinli) [reply ccpp/data/GFS_typedefs.F90:5624]: Yes, the are set = true in the namelist.

I agree with. In order to save time, we keep the current status of "kdvel",  "EMIS_ANT_NO" and "vdep" this time.
We will get rid of "kdvel",  "EMIS_ANT_NO" and "vdep" in the next commit.

File: ccpp/data/GFS_typedefs.meta [hunk 2]
```diff
@@ -620,6 +634,27 @@
...
+  dimensions = (horizontal_dimension,1)
+  type = real
+  kind = kind_phys
+[smoke_GBBEPx]
+  standard_name = emission_smoke_GBBEPx
+  long_name = emission fire GBBEPx
+  units = various
+  dimensions = (horizontal_dimension,24,3)
+  type = real
+  kind = kind_phys
```
Comment (SamuelTrahanNOAA) [inline ccpp/data/GFS_typedefs.meta]: You're missing the `active=` lines for these and many other variables. The non-smoke configurations will fail unless you have `active=` lines for variables that are not always allocated.

**Pull request merged**